include/cppkafka/utils/buffered_producer.h

```diff
@@ -197,11 +207,28 @@ class CPPKAFKA_API BufferedProducer {
     size_t get_buffer_size() const;
     
     /**
-     * \brief Returns the total number of messages ack-ed by the broker since the beginning
+     * \brief Get the number of messages not yet acked by the broker
      *
      * \return The number of messages
      */
-    size_t get_total_messages_acked() const;
+    size_t get_pending_acks() const;
+    
+    /**
+     * \brief Get the total number of messages successfully produced since the beginning
+     *
+     * \return The number of messages
+     */
+    size_t get_total_messages_produced() const;
+    
+    /**
+     * \brief Get the total outstanding flush operations in progress
+     *
+     * Since flush can be called from multiple threads concurrently, this counter indicates
+     * how many operations are curretnly in progress.
+     *
+     * \return The number of outstanding flush operations.
+     */
+    size_t get_flushes_in_progress() const;
```

Comment:
Well the idea is well explained in the header of the class. By itself it's not too useful, you have to pair it with `get_buffer_size()` which if both return 0 is a good indication that everything has been pushed to the broker. The use case is that I'm writing a library on top of yours with a much higher level of abstraction (i.e. imagine dozens of producers and consumers maybe more), and I'm taking in messages in native format, serialize them and buffer them in chunks because I'm pulling them from application queues, and then flush them so I need a deterministic way of knowing when to delete the original (non-serialized) data - held in unique ptrs. If it turns out that it's not a reliable process or if I change my design later on, I will prob have some sort of correlation hash map based on the message handle and use that along with the success callback to delete each message as it's acked. But I kinda want to avoid all the look-ups. 
Otherwise, it may be good to wait for all flushes to end before shutting down the producer for example.

Reply:
So to answer:
> You can't really make a decision based on this as by the time the function returns, any in-progress flush may have already finished

Exactly and that's fine. I'm checking when flushes get to 0, I don't care about how many are at any point in time.